backend/routes/data.py

```diff
@@ -22,6 +23,59 @@ def send_audio_file(file_name):
     return send_from_directory(app.config["UPLOAD_FOLDER"], file_name)
 
 
+def validate_segmentation(segment):
+    """Validate the segmentation before accepting the annotation's upload from users
+    """
+    required_key = {"annotations", "end_time", "transcription"}
+
+    if set(required_key).issubset(segment.keys()):
+        return True
+    else:
+        return False
+
+
+def generate_segmentation(
+    annotations, transcription,
+    start_time, end_time, data_id=None, segmentation_id=None
+):
+    """Generate a Segmentation from the required segment information
+    """
+    if segmentation_id is None:
+        if data_id is None:
+            segmentation = Segmentation(
+                start_time=start_time, end_time=end_time
+            )
+        else:
+            segmentation = Segmentation(
+                data_id=data_id,
+                start_time=start_time, end_time=end_time
+            )
+    else:
+        segmentation = Segmentation.query.filter_by(
+            data_id=data_id, id=segmentation_id
+        ).first()
+
+    segmentation.set_transcription(transcription)
+    values = []
+    for label_name, label_value in annotations.items():
+        app.logger.info(label_name)
+        app.logger.info(label_value)
+        if type(label_value["values"]) is list:
+            for val_id in label_value["values"]:
+                value = LabelValue.query.filter_by(
+                    id=int(val_id), label_id=label_value["label_id"]
+                ).first()
+                values.append(value)
+        else:
+            value = LabelValue.query.filter_by(
+                id=int(label_value["values"]), label_id=label_value["label_id"]
+            ).first()
+            values.append(value)
```

Comment:
This logic will fail as I expect users to pass label `values` and not `ids`. I don't expect users to remember label ids and label value ids and neither do we show them on the UI. For now, I think we should show actual `label_id` as well as `label_value_id` on the user interface of `Label` dashboard as well as `Label Value` dashboard for this.

Reply:
This can go under table header `#`.

Reply:
This logic was copied from `add_segmentations` function:
https://github.com/midas-research/audino/blob/3de7bc9de592b2ceb897b8c3846a3f8ecb68c89e/backend/routes/projects.py#L580